src/compiler/parse/read/context.ts

```diff
@@ -116,8 +117,11 @@ export default function read_context(parser: Parser) {
 				break;
 			}
 
+			// TODO: DRY this out somehow
+			// We don't know whether we want to allow reserved words until we see whether there's a ':' after it
+			// Probably ideally we'd use Acorn to do all of this
```

Comment:
I'm really not sure how achievable this ideal is. Walking through the AST to confirm that it only contains allowable destructurings doesn't sound too impossible, but I don't immediately see how we'd stop Acorn from parsing keyed each blocks as a function call - `{#each a as b (c)}`.